spring-boot-modules/spring-boot-crud/src/main/java/com/baeldung/entity_dto_differences/dto/BookDto.java

```diff
@@ -0,0 +1,35 @@
+package com.baeldung.entity_dto_differences.dto;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.validation.constraints.NotNull;
+
+public class BookDto {
+
+    @NotNull
+    @JsonProperty("NAME")
```

Comment:
These custom JSON field names are untested. In your integration test, you should assert on the string and not reuse an ObjectMapper. I can edit this to @JsonProperty("WRONG") and none of the tests fail.